examples/sim_fibo_ad_cp.cpp

```diff
@@ -206,6 +207,38 @@ try
                                   *props, deck, gravity[2], state);
     }
 
+    std::shared_ptr<BlackoilState> distributed_state=Dune::stackobject_to_shared_ptr(state);
```

Comment:
The problem is that we need to pass the correct state to simulator in https://github.com/blattms/opm-autodiff/blob/60cdc58/distribute-input-props-data/examples/sim_fibo_ad_cp.cpp#L300 There we pass the distributed_state to the simulator. In sequential runs this is the original object, otherwise the distributed. To prevent copies  is use the stackobject_to_shared_ptr. In the parallel case the smart pointer gets reset to one with the distributed state and a default deleter. No lifetime issues are introduced that have not been there before.